x/ecocredit/server/basket/query_balance.go

```diff
@@ -16,12 +19,18 @@ func (k Keeper) BasketBalance(ctx context.Context, request *baskettypes.QueryBas
 
 	basket, err := k.stateStore.BasketTable().GetByBasketDenom(ctx, request.BasketDenom)
 	if err != nil {
-		return nil, err
+		if ormerrors.IsNotFound(err) {
+			return nil, sdkerrors.Wrapf(err, "basket %s not found", request.BasketDenom)
+		}
+		return nil, sdkerrors.Wrapf(err, "failed to get basket %s", request.BasketDenom)
 	}
 
```

Comment:
One more check here to see if the batch exists might be nice.

```suggestion

	found, err := k.coreStore.BatchTable().HasByDenom(ctx, request.BatchDenom)
	if err != nil {
		return nil, sdkerrors.Wrapf(err, "failed to get credit batch %s", request.BatchDenom)
	}
	if !found {
		return nil, ormerrors.NotFound.Wrapf("credit batch %s not found", request.BatchDenom)
	}

```

Reply:
Current result if batch does not exist:

```
./build/regen q ecocredit basket-balance eco.uC.NCT Z99-999-99999999-99999999-999
```

```
balance: "0"
```

Suggested result if batch does not exist:

```
./build/regen q ecocredit basket-balance eco.uC.NCT Z99-999-99999999-99999999-999
```

```
Error: rpc error: code = InvalidArgument desc = credit batch Z99-999-99999999-99999999-999 not found: not found: invalid request
```

Reply:
You'll have to insert the batch in the test for tests to pass:

```
	coreapi "github.com/regen-network/regen-ledger/api/regen/ecocredit/v1"
```

```
	err = s.coreStore.BatchTable().Insert(s.ctx, &coreapi.Batch{
		Denom: batchDenom,
	})
	require.NoError(t, err)
```